packages/react-reconciler/src/ReactFiberDevToolsHook.js

```diff
@@ -65,12 +53,43 @@ export function injectInternals(internals: Object): boolean {
   try {
     const rendererID = hook.inject(internals);
     // We have successfully injected, so now it is safe to set up hooks.
-    onCommitFiberRoot = catchErrors(root =>
-      hook.onCommitFiberRoot(rendererID, root),
-    );
-    onCommitFiberUnmount = catchErrors(fiber =>
-      hook.onCommitFiberUnmount(rendererID, fiber),
-    );
+    onCommitFiberRoot = (root, expirationTime) => {
+      try {
+        if (enableProfilerTimer) {
+          const currentTime = requestCurrentTime();
+          const priorityLevel = inferPriorityFromExpirationTime(
+            currentTime,
+            expirationTime,
+          );
```

Comment:
Andrew mentioned maybe we could use [`getCurrentPriorityLevel()`](https://github.com/facebook/react/blob/9c6de716d028f17736d0892d8a3d8f3ac2cb62bd/packages/react-reconciler/src/SchedulerWithReactIntegration.js#L87-L102) here rather than inferring the priority level. I don't _think_ the priority level a Fiber was scheduled with will necessarily match the current priority level though. (Maybe I misunderstood his suggestion.)

Reply:
If you call `getCurrentPriorityLevel` inside `commitRoot` (not `commitRootImpl`, since that's always run at `ImmediatePriority` — it has to be the wrapper) then it will match the priority of the commit, which is not necessarily the same thing as the priority at which something was scheduled because multiple priorities can get batched.

This approach seems fine, though.

It's possible we may eventually track the priority on the update object.